Baboon_stuff/Baboon_Reader.cpp

```diff
@@ -1,77 +1,152 @@
+    {
+      while ( getline (myfile,line) ) //assigns each line of the file to the string line.
+      {
+        stringstream s_stream(line);// don't quite understand this yet. The documentation was a little confusing, will read further.
+        vector<string> result;
+        while(s_stream.good()) //goes through each line (while there are lines to go through) and seperates based on the delimeter (the comma)
+        {
+          string substr;
+          getline(s_stream, substr, ',');
+          result.push_back(substr);
+        }
+        if (find(baboon_names.begin(), baboon_names.end(),result[8]) != baboon_names.end()){
+          ;
+        }else{
+          baboon_names.push_back(result[8]); //checks to see if baboon name is already present in the baboon list. If not it creates an empty vector that will be added to and adds to the baboon list.
+        }
+      return baboon_names;
+    }
     }
-    myfile.close();
+  }
 
+  map<string, double> jaccard_index_for_each(map<string, vector<tuple<double,double,double>>> dicty, string baboon_name ){
     /*
-    If you are wanting to print it out.
-    for (pair<string, vector<string>> element : baboon_personal_data) {
-    string key = element.first;
-    vector<string> value = element.second;
-    cout << key <<"::"<< endl;
-    for (int i=0;i<value.size();i++){
-      cout<<value[i]<<endl;
+    0.0001° = 11.1 m
+    0.00001° = 1.11 m
+    This function aims to generate the jaccard index between the baboon of interest and every other baboon listed.
+    This is most likely not the most efficient way to measure fly by but it's good practice nonetheless.
+
+    The timestamps are out of sync but I should see whether the number of timestamps are consistant.
+    In this instantiation I pretend that the timestamps are in sync and that the number of timestamps are consistant.
+
+    I think I may have to search for a new data set as the timestamps are not very consistant in their sampling.
+    I think the ideal data set would have at least one of the two characteristics:
+    1. large sampling frequency (second or minute based rather then by hour)
+    2. consistant sampling (sampling at the same times for each baboon)
+    */
+
+    vector<string> baboon_names=baboonNamesList();
+    map<string, double> dictToReturn;
+    double longitude_tolerance =0.00001;
+    double latitude_tolerance =0.00001;
+    for (int i = 0; i < dicty[baboon_name].size(); i++) {
+      for(int j = 0; j< baboon_names.size();j++){
+        double totalIntersections = 0;
+        for(int k =0; k< dicty[baboon_names[j]].size();k++){
+          if (get<1>(dicty[baboon_name][i])-longitude_tolerance <= get<1>(dicty[baboon_names[j]][k]) &&get<1>(dicty[baboon_names[j]][k])<=get<1>(dicty[baboon_name][i])+longitude_tolerance){
+            if (get<2>(dicty[baboon_name][i])-latitude_tolerance <= get<2>(dicty[baboon_names[j]][k]) &&get<2>(dicty[baboon_names[j]][k])<=get<2>(dicty[baboon_name][i])+latitude_tolerance){

```

Comment:
This is a great example of where you might benefit from some _refactoring_: can you transform these three lines into just one function call?

Reply:
I know I can combine the two if statements as with && it will only evaluate the right side if the left side is true. With regards to transforming the lines into a function call I assume you mean taking the mess of nested for loops and declaring them as their own function and then just calling it here? Or is there some better function used in c++ that I am missing? I also just recently watched a video on list comprehension, it seems to make the syntax more concise. Would that qualify as a form of refactoring? If so I can work on implementing that in place of these burdensome for loops.